src/dovi/rpu_info.rs

```diff
@@ -333,17 +382,248 @@ impl RpusListSummary {
+                            format!("{area}: ({min} - {max})")
+                        }
+                    })
+            })
+            .join(", ");
+
+        let l8_trims = if dmv2_count > 0 {
+            let l8_trims_str: Vec<String> = rpus
+                .iter()
+                .filter_map(|rpu| {
+                    rpu.vdr_dm_data.as_ref()?.get_block(8).and_then(|block| {
+                        if let ExtMetadataBlock::Level8(l8) = block {
+                            Some(l8.trim_target_nits())
+                        } else {
+                            None
+                        }
+                    })
+                })
+                .unique()
+                .sorted()
+                .map(|target_nits| format!("{target_nits} nits"))
+                .collect();
+
+            Some(l8_trims_str)
+        } else {
+            None
+        };
+
+        let l9_mdp = if dmv2_count > 0 {
+            let l9_mdp_str: Vec<String> = rpus
+                .iter()
+                .filter_map(|rpu| {
+                    rpu.vdr_dm_data.as_ref()?.get_block(9).and_then(|block| {
+                        if let ExtMetadataBlock::Level9(l9) = block {
+                            Some(l9.source_primary_index)
+                        } else {
+                            None
+                        }
+                    })
+                })
+                .fold(HashMap::new(), |mut frames, idx| {
+                    *frames.entry(idx).or_insert(0) += 1;
+                    frames
+                })
+                .into_iter()
+                .sorted_by_key(|e| e.0)
+                .map(|(idx, frames)| {
+                    let alias = MasteringDisplayPrimaries::from(idx).to_string();
+                    if frames < dmv2_count {
+                        format!("{alias} ({frames})")
+                    } else {
+                        alias
+                    }
+                })
+                .collect();
+
+            Some(l9_mdp_str)
+        } else {
+            None
+        };
+
         Ok(Self {
             count: rpus.len(),
             scene_count,
             rpu_mastering_meta_str,
             profiles_str,
             dm_version_str,
             dm_version_counts,
+            dmv2,
             l6_meta,
+            l5_str,
+            l8_trims,
+            l9_mdp,
             l1_data,
             l1_stats,
             l2_trims,
+            l2_data: None,
+            l2_stats: None,
+            l8_data: None,
+            l8_stats_trims: None,
+            l8_stats_saturation: None,
+            l8_stats_hue: None,
         })
     }
+
+    pub fn with_l2_data(rpus: &[DoviRpu]) -> Result<Self> {
+        let mut summary = Self::new(rpus)?;
+
+        let l2_data: Vec<_> = rpus
+            .iter()
+            .map(|rpu| {
+                if let Some(ExtMetadataBlock::Level2(l2)) =
+                    rpu.vdr_dm_data.as_ref().and_then(|dm| dm.get_block(2))
```

Comment:
I'm pretty sure `get_block` only returns the first trim block.
There needs to be a way to specify the target nits but ATM I'm not sure the crate has a convenient API for it.

It's probably fine for now.